src/runtime/virtcontainers/sandbox.go

```diff
@@ -1234,9 +1237,24 @@ func (s *Sandbox) startVM(ctx context.Context) (err error) {
 		return err
 	}
 
+	if prestartHookFunc != nil {
+		hid, err := s.GetHypervisorPid()
+		if err != nil {
+			return err
+		}
+		s.Logger().Infof("hypervisor pid is %v", hid)
+		ctx = context.WithValue(ctx, HypervisorPidKey{}, hid)
+
+		if err := prestartHookFunc(ctx); err != nil {
+			return err
+		}
+	}
+
 	// In case of vm factory, network interfaces are hotplugged
 	// after vm is started.
-	if s.factory != nil {
+	// In case of prestartHookFunc, network config might have been changed.
+	// We need to rescan and handle the change.
+	if s.factory != nil || prestartHookFunc != nil {
```

Comment:
Seems the prestartHookFunc will never be nil in this function, although there is not necessarily a real “prestart hook” (which means even if the prestart hook is not actually executed, we will always scan the netns once).
I think the program can run without errors, but I'm not sure whether this is as expected.

Reply:
From VC API point of view, IMO it's better to check for nil `prestartHookFunc` although right now the caller always sets it.

Reply:
Indeed it is.